### ISSUE TITLE
vue/no-unused-components did not support jsx

### ISSUE THREAD

**Issue by 374632897** [vuejs/eslint-plugin-vue#1108]
**Tell us about your environment**	
* **ESLint version:**  5.16.0
* **eslint-plugin-vue version:**	5.2.3
* **Node version:**	 10.14.2

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  'extends': [
    'plugin:vue/recommended',
    '@vue/prettier',
    '@nibfe/eslint-config',
    'plugin:jest/recommended',
  ],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    '@getify/proper-ternary/where': 'off',
    '@getify/proper-ternary/parens': 'off',
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
}

```

**What did you do?**	

```vue
<template>
   <div></div>
</template>

<script>
import { mapState } from 'vuex'
export default {
  name: '',
  components: {
      HelloWorld,
  },
  data() {
    return {}
  },

  watch: {},
  computed: {},
  mounted() {
       this.showDialog()
  },
  methods: {
      showDialog() {
           this.$confirm({
              message: <HelloWorld />
          })
     }
  },
}
</script>
```

**What did you expect to happen?**
The component HelloWorld will pass the test.

**What actually happened?**

The "HelloWorld" component has been registered but not used (vue/no-unused-components)

**Comment by ota-meshi** [comment]
Hi @374632897,

Perhaps you don't need to define components.

```diff
export default {
  name: '',
-  components: {
-      HelloWorld,
-  },
// ...
```

**Comment by 374632897** [comment]
> Hi @374632897,
> 
> Perhaps you don't need to define components.
> 
> ```diff
> export default {
>   name: '',
> -  components: {
> -      HelloWorld,
> -  },
> // ...
> ```

@ota-meshi If I don't register the component in components, Vue will throw a warning like this: 

```
[Vue warn]: Unknown custom element: <hello-world> - did you register the component correctly? For recursive components, make sure to provide the "name" option.
```

I've seen the source code of eslint-plugin-vue，It only handles the component used in template, and the components used dynamically by jsx will be ignored.

**Comment by ota-meshi** [comment]
I looked at the following link and thought that I didn't need the `components` definition.

https://vuejs.org/v2/guide/render-function.html#JSX

I'm not familiar with using Vue and JXS together.
Can you share a minimal repository that reproduces this issue?

**Comment by 374632897** [comment]
I tried to make a demo，then I found it works as expected without define a component in components，the error happened because I use PascalCase and kebab-case mixed.
I am so sorry for taking your time because of my mistake. I'll close this issue. Thank you very much.
